src/asyncLocalStorage.ts

```diff
@@ -0,0 +1,31 @@
+interface IAsyncStorage {
```

Comment:
should this be called `IAsyncLocalStorage`? It only supports strings, so it's not generic enough and is potentially confusing to be called as such (initially was thinking of making it more generic, but should still have a more specific version defined too)